pkgs/tools/archivers/7zz/default.nix

```diff
@@ -2,23 +2,27 @@
 , lib
 , fetchurl
 
+  # Only used for x86/x86_64
 , uasm
-, useUasm ? stdenv.isLinux
+, useUasm ? stdenv.hostPlatform.isx86
 
   # RAR code is under non-free unRAR license
   # see the meta.license section below for more details
 , enableUnfree ? false
+
+  # For tests
+, _7zz
+, testVersion
 }:
 
 let
   inherit (stdenv.hostPlatform) system;
-  platformSuffix =
-    lib.optionalString useUasm {
-      aarch64-linux = "_arm64";
-      i686-linux = "_x86";
-      x86_64-linux = "_x64";
-    }.${system} or
-      (builtins.trace "7zz's ASM optimizations not available for `${system}`. Building without optimizations." "");
+  platformSuffix = {
+    aarch64-linux = "_arm64";
+    i686-linux = "_x86";
+    x86_64-linux = "_x64";
+  }.${system} or
+    (builtins.trace "`platformSuffix` not available for `${system}.` Making a generic `7zz` build." "");
```

Comment:
Is this clarification in the wording because the generic build can have ASM optimizations too?
The profile specific make files do have `USE_ASM=1`. If we want the makefiles to be detached from whether we use ASM or not we probably want to control `USE_ASM` based on `useUasm`

Reply:
No, the generic build doesn't have ASM optimizations. All the other ones (the architecture specific ones) seems to have, however it is not a simple case of setting `USE_ASM` because some architectures use third party tools to generate the assembly code (e.g. `x86_64` uses `uasm`), while others doesn't (e.g.: `aarch64-linux`).

Reply:
ok cool so `USE_ASM` is general and `useUasm` is for using `usam`

:+1: